### PR TITLE
Warn when selected rx protocol is not in build configuration

### PR DESCRIPTION
**Pull request by her01n**
User can select a receiver protocol when upgrading the firmware. Later, the protocol can be selected in the receiver tab. Only the protocol selected at the firmware upgrade time would work. Let's warn the user if the protocol selected in the receiver tab will not work.

### CODE REVIEW

**Comment by haslinghuis** [comment]
Instead of a warning unavailable rx protocols should be hidden.

**Comment by her01n** [comment]
Unsupported rx protocols are disabled. I have retained the warning - it may be displayed when wrong protocol is already selected by previous configurator version.

**Comment by haslinghuis** [comment]
image

**Review by haslinghuis — APPROVED** [review]
LGTM

**Comment by nerdCopter** [comment]
good work!
why would `target_custom` be enabled is i flashed only CRSF?
image

**Comment by nerdCopter** [comment]
looks good with a few multiple protocols added as well
image

i just need to know why `target_custom` is available, or if it is to be expected before i approve.

**Comment by her01n** [comment]
TARGET_CUSTOM has no corresponding build option, as listed in src/js/fc.js:72. I guess it is always available.

**Comment by haslinghuis** [comment]
There exists `USE_SERIALRX_TARGET_CUSTOM`, so prefer this be handled like other cases as it probably is to be used as custom define anyway.

File: src/js/fc.js [hunk 1]
```diff
@@ -795,6 +795,52 @@ const FC = {
         return serialRxTypes;
     },
 
+    getSupportedSerialRxTypes: () => {
+        if (FC.CONFIG.buildOptions?.length) {
+            const options = FC.CONFIG.buildOptions;
+            let supportedRxTypes = ['NONE', 'TARGET_CUSTOM'];
```
Comment (nerdCopter) [inline src/js/fc.js:801]:
```suggestion
            let supportedRxTypes = ['NONE',];
            if (options.includes('USE_SERIALRX_TARGET_CUSTOM')) {
                supportedRxTypes.push('TARGET_CUSTOM');
            }  
```

**Review by nerdCopter — CHANGES_REQUESTED** [review]

File: src/js/fc.js (hunk 1 above)
Comment (nerdCopter) [reply src/js/fc.js:801]: @haslinghuis , please verify, TY.

**Comment by haslinghuis** [comment]
image

**Review by nerdCopter — APPROVED** [review]

**Comment by nerdCopter** [comment]
i just noticed if a pilot pastes a `diff` that contains a non-active protocol, the UI still shows the configured protocol in normal font shade (black here).
i wonder if there is a way to dimm (gray) the set protocol if the user did not cloud-build them.

e.g. i built the firmware with CRSF, but the quad is SBUS, i pasted my diff and went to the receiver tab to see the SBUS settings in normal color even though the firmware does not contain SBUS.  yes, user-error, but maybe a mitigable situation.
image

**Comment by haslinghuis** [comment]
- think this should be fixed at the source (firmware).

image

- alternative is to set to NONE when invalid provider is used.

image

**Comment by nerdCopter** [comment]
It does already says `not supported` but my eyes skipped over such.  Either way is fine to me. Seemingly a dimm font would be a good visual-note and maybe easy to factor. :man_shrugging:.
In this case, I happened to wonder why RX/TX was not working, then realized why (i'm accustomed to CRSF now).  I commented because i think other pilots might experience same as well.
Of course this idea is not all that important at all.

**Comment by nerdCopter** [comment]
It does already says `not supported` but my eyes skipped over such.  Either way is fine to me. Seemingly a dimm font would be a good visual-note and maybe easy to factor. :man_shrugging:.
In this case, I happened to wonder why RX/TX was not working, then realized why (i'm accustomed to CRSF now).  I commented because i think other pilots might experience same as well.
Of course this idea is not all that important at all.

edit: occurs to me that #2 (`NONE`) might not accurately reflect is CLI setting.

**Comment by haslinghuis** [comment]
image

**Pull request merged**